pkg/controller/v1alpha2/experiment/util/client_util.go

```diff
@@ -16,14 +16,118 @@ limitations under the License.
 package util
 
 import (
+	"context"
+	"io/ioutil"
+	"strings"
+
+	experimentv1alpha2 "github.com/kubeflow/katib/pkg/api/operators/apis/experiment/v1alpha2"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	cl "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
-func NewClient(options client.Options) (client.Client, error) {
+type katibClient struct {
+	client client.Client
+}
+
+const (
+	trialTemplatesName            = "trial-template"
+	metricsCollectorTemplatesName = "metricscollector-template"
+)
+
+func NewClient(options client.Options) (*katibClient, error) {
 	cfg, err := config.GetConfig()
 	if err != nil {
 		return nil, err
 	}
-	return client.New(cfg, options)
+	cl, err := client.New(cfg, options)
+	return &katibClient{
+		client: cl,
+	}, nil
+}
+
+func (k *katibClient) GetExperimentList(namespace ...string) (*experimentv1alpha2.ExperimentList, error) {
+	ns := getNamespace(namespace...)
+	expList := &experimentv1alpha2.ExperimentList{}
+	listOpt := cl.InNamespace(ns)
+
+	if err := k.client.List(context.Background(), listOpt, expList); err != nil {
+		return expList, err
+	}
+	return expList, nil
+
+}
+
+func (k *katibClient) CreateExperiment(experiment *experimentv1alpha2.Experiment, namespace ...string) error {
+
+	if err := k.client.Create(context.Background(), experiment); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k *katibClient) GetTrialTemplates(namespace ...string) (map[string]string, error) {
+
+	ns := getNamespace(namespace...)
+	trialTemplates := &apiv1.ConfigMap{}
+
```

Comment:
Suggest move getConfigMap into this file, too and then please call getConfigMap

Reply:
Done